### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by glasser**
- Remove a file that's not in the repo anymore
- Switch docs to team rather than list of users

### CODE REVIEW

**Comment by glasser** [comment]
I'm not sure if the point of the docs line is primarily around access control or setting up default reviewers for PRs; maybe it makes sense to keep as a list of people (but probably add me?) rather than switching to the team?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,9 +1,4 @@
 # The renovate-approve bot is allowed to merge its own PRs.  As it's not
 # a regular GitHub user (but rather, an app!), its username is indicated
 # as such, using the `app/` prefix. https://github.com/apps/renovate-approve
-/docs/ app/renovate-approve @stephenbarlow @abernix @trevor-scheer
-
-# Changes to the federation specification should be further reviewed.
-# Most importantly because they require messaging and awareness to
-# other implementations other than our own.
-/docs/source/federation/federation-spec.md @jbaxleyiii
+/docs/ app/renovate-approve @stephenbarlow @apollographql/polaris
```
Comment (abernix) [inline .github/CODEOWNERS:4]:
```suggestion
/docs/ app/renovate-approve @stephenbarlow @glasser
```

**Review by abernix — APPROVED** [review]
I'd suggest just you for time time-being, @glasser. (+ otherwise LGTM)

**Comment by kitikhunp** [comment]
%

**Pull request merged**